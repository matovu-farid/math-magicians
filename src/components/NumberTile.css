.tile {
  background-color: #e0e0e0;
  border: 1px solid black;
  padding: 20px;
  text-align: center;
  max-width: 50px;
}