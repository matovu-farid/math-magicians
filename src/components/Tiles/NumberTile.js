import { Component } from 'react';
import PropTypes from 'prop-types';
import './Tile.css';

export default class NumberTile extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { value } = this.props;
    return (<div className="tile">{value}</div>);
  }
}
NumberTile.propTypes = {
  value: PropTypes.string.isRequired,
};
