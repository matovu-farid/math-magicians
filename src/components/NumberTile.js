import './NumberTile.css';

const NumberTile = () => <div className="tile">3</div>;
export default NumberTile;
