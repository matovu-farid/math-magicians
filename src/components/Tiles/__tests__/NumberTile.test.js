import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NumberTile from '../NumberTile';

it('renders correctly', () => {
  const tree = renderer.create(<NumberTile value={7} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display correct value', () => {
  render(<NumberTile value={7} />);
  const button = screen.queryByRole('button');

  expect(button).toHaveTextContent('7');
});
