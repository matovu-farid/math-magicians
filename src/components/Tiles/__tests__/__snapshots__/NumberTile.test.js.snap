// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="tile"
  onClick={[Function]}
  type="button"
>
  7
</button>
`;
